client/src/components/addSourceWizard/addSourceWizardStepOne.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Form, Radio } from 'patternfly-react';
+import Store from '../../redux/store';
+import helpers from '../../common/helpers';
+import { apiTypes } from '../../constants';
+import { sourcesTypes } from '../../redux/constants';
+import _ from 'lodash';
+
+class AddSourceWizardStepOne extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.initialState = {
+      source: {},
+      stepOneValid: false,
+      sourceType: '',
+      sourceTypeError: null
+    };
+
+    this.state = { ...this.initialState, ...this.resetInitialState(props) };
+
+    helpers.bindMethods(this, ['onChangeSourceType']);
+  }
+
+  resetInitialState(nextProps) {
+    return {
+      source: nextProps.source || {},
+      sourceType: nextProps.source[apiTypes.API_SOURCE_TYPE] || ''
+    };
+  }
+
+  /*
+  componentWillReceiveProps(nextProps) {
+    // this.resetInitialState(nextProps);
+  }
+
+  resetInitialState(nextProps) {
+    if (
+      nextProps.source &&
+      this.state.sourceType !== nextProps.source[apiTypes.API_SOURCE_TYPE]
+    ) {
+      this.setState(
+        Object.assign(
+          { ...this.initialState },
+          {
+            source: nextProps.source,
+            sourceType: nextProps.source[apiTypes.API_SOURCE_TYPE]
+          }
+        ),
+        () => this.validateStep()
+      );
+    }
+  }
+  */
+
+  validateStep() {
+    const { sourceType, sourceTypeError, source } = this.state;
+
+    if (!sourceTypeError) {
+      Store.dispatch({
+        type: sourcesTypes.UPDATE_SOURCE_WIZARD_STEPONE,
+        source: _.merge({}, source, { [apiTypes.API_SOURCE_TYPE]: sourceType })
+      });
+    }
+  }
+
+  validateSourceType(sourceType) {
+    if (sourceType === '') {
```

Comment:
would this be possible?

Reply:
Oddly, currently, yes since it doesn't default to any one radio button value. If we select one, then no, that check doesn't need to exist.

Reply:
we should default to Network, IMO

Reply:
Removed import, and defaulted to network

Reply:
code is still here.